doc/cprover-manual/contracts-assigns.md

```diff
@@ -1,50 +1,144 @@
 [CPROVER Manual TOC](../../)
 
-# Assigns Clause
+# Assigns Clauses
 
-## In Function Contracts
+An _assigns_ clause lets the user to specify which memory locations may be
+assigned to by a function or a loop.
 
-### Syntax
+The set of locations described by the contract is the union of the locations
+specified by each of its assigns clauses, or the empty set of no _assigns_
+clause is specified.
+
+The assigns clause may only refer to locations that exist in the calling context
```

Comment:
If a function is called in exactly two places, and both of them have a procedure-local variable `x` in scope, can I use `&x` in the assigns clause?

Reply:
syntactically the targets must be expressions of function parameters or globals, but they can surely resolve to that procedure local `&x` at some call-site. However I'm not sure what it would mean to modify the address of a procedure local variable. Modifying the value stored at this address, yes.

Reply:
Apologies, I should have used `x` instead of `&x`. I don't see where the documentation makes explicit that such expressions must be function parameters or globals?